Automattic/wp-calypso
# A4A: add Needs setup menu item when necessary

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/374

## Proposed Changes

Note: this is to replace PR here https://github.com/Automattic/wp-calypso/pull/90065

This PR adds a new menu item to the sidebar Needs setup. The link is currently temporary and logic will be developed in the future.

## Testing Instructions



- Spin up this branch locally
- Start with no wpcom hosting purchased.
- Navigate to `/sites`. 3 Original menu items should appear.

<img width="246" alt="Screenshot 2024-04-30 at 11 15 06 AM" src="https://github.com/Automattic/wp-calypso/assets/60262784/a4760bb0-673c-47cf-88e7-5ba87b962117">

- Navigate to `/marketplace/hosting` and purchase WPCOM hosting
- Navigate to `/sites`. 4 menu items should appear.

<img width="238" alt="Screenshot 2024-04-30 at 11 08 21 AM" src="https://github.com/Automattic/wp-calypso/assets/60262784/e4468cde-64e2-414b-b8bc-0a1d127541aa">

## Pre-merge Checklist
